src/ensembl/src/content/app/browser/browser-region-editor/BrowserRegionEditor.test.tsx

```diff
@@ -0,0 +1,181 @@
+import faker from 'faker';
+
+import { BrowserRegionEditor } from './BrowserRegionEditor';
+import Input from 'src/shared/components/input/Input';
+import Select from 'src/shared/components/select/Select';
+import Tooltip from 'src/shared/components/tooltip/Tooltip';
+
+import { ChrLocation } from '../browserState';
+import genomeKaryotypes from 'tests/data/browser/karyotypes';
+import { getCommaSeparatedNumber } from 'src/shared/helpers/numberFormatter';
+import { LoadingState } from 'src/shared/types/loading-state';
+import { createValidationInfo } from 'tests/fixtures/browser';
+
+const defaultProps = {
+  activeGenomeId: faker.lorem.words(),
+  chrLocation: ['13', 2315086, 32400266] as ChrLocation,
+  genomeKaryotypes,
+  regionEditorActive: false,
+  regionFieldActive: false,
+  regionValidationInfo: {},
+  regionValidationLoadingStatus: LoadingState.NOT_REQUESTED,
+  changeBrowserLocation: jest.fn(),
+  replace: jest.fn(),
+  resetRegionValidation: jest.fn(),
+  toggleRegionEditorActive: jest.fn(),
+  validateRegion: jest.fn()
+};
+
+describe('<BrowserRegionEditor', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  let wrapper: any;
+
+  beforeEach(() => {
+    wrapper = mount(<BrowserRegionEditor {...defaultProps} />);
+  });
+
+  describe('rendering', () => {
+    test('contains Select', () => {
+      expect(wrapper.find(Select).length).toBe(1);
+    });
+
+    test('contains two Input elements', () => {
+      expect(wrapper.find(Input).length).toBe(2);
+    });
+
+    test('contains submit and close buttons', () => {
+      expect(wrapper.find('.submitButton')).toHaveLength(1);
+      expect(wrapper.find('.closeButton')).toHaveLength(1);
+    });
+
+    test('has an overlay on top when region field is active', () => {
+      wrapper.setProps({ regionFieldActive: true });
+      expect(wrapper.find('.browserOverlay').length).toBe(1);
+    });
+  });
+
+  describe('behaviour', () => {
+    const locationStartInput = getCommaSeparatedNumber(10);
+    const locationEndInput = getCommaSeparatedNumber(10500);
+
+    test('shows form buttons when focussed', () => {
+      wrapper.find(Select).simulate('focus');
+      expect(wrapper.props().toggleRegionEditorActive).toHaveBeenCalledTimes(1);
+    });
+
+    test('applies correct value on change', () => {
+      // TODO:
+      // Couldn't find a good way to test this for <Select />. So write tests once it is figured out.
+
+      wrapper
+        .find(Input)
+        .first()
+        .simulate('change', { target: { value: locationStartInput } });
+
+      expect(
+        wrapper
+          .find(Input)
+          .first()
+          .props().value
+      ).toBe(locationStartInput);
+
+      wrapper
+        .find(Input)
+        .last()
+        .simulate('change', { target: { value: locationEndInput } });
+
+      expect(
+        wrapper
+          .find(Input)
+          .last()
+          .props().value
+      ).toBe(locationEndInput);
+    });
```

Comment:
I am not sure this test is interesting.
Testing that when a start input and an end input are filled and a chromosome selected, and the apply button is pressed, then something happens — now that would be interesting.

Reply:
Removed these. The tests for validation and submission are already available.